### PR TITLE
Add com.arteeh.Flasher

### PR DESCRIPTION
**Pull request by arteeh**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

This is my first Linux desktop app, GTK app and Flatpak app, so I might have made mistakes. Help is much appreciated.

### CODE REVIEW

**Comment by arteeh** [comment]
bot, build com.arteeh.Flasher

**Comment by arteeh** [comment]
bot, build com.arteeh.Flasher

**Comment by arteeh** [comment]
bot, build com.arteeh.Flasher

**Comment by arteeh** [comment]
Note to self:

- Software can't find the screenshots, maybe they should be jpg instead of png? Maybe host them somewhere other than the repo?
- Description is wrong: 0.1 doesn't support wasp-bootloader
- In Software release 0.1 is allegedly from 12 hours ago. Should I enter the time somewhere?

**Comment by arteeh** [comment]
Journalctl log when I go to my app page:

`gnome-software[1886]: Result of screenshot downloading attempt with status code '404': Not Found`

PNG is recommended, and the link from my metainfo does work in the browser. Am I supposed to host it somewhere without HTTPS?

File: com.arteeh.Flasher.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+app-id: com.arteeh.Flasher
+# Libhandy depends on Vala, so the freedesktop platform is not enough
+runtime: org.gnome.Platform
+runtime-version: '3.38'
```
Comment (barthalion) [inline com.arteeh.Flasher.yml:4]:
```suggestion
runtime-version: '40'
```

File: com.arteeh.Flasher.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+sdk: org.gnome.Sdk
+command: /app/bin/pinetime-flasher
```
Comment (barthalion) [inline com.arteeh.Flasher.yml:6]:
```suggestion
command: pinetime-flasher
```

It's a nitpick, but technically `/app/bin` is already in the $PATH inside the sandbox so no need to use absolute path.

File: com.arteeh.Flasher.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  # GTK widgets for mobile
+  - name: libhandy
+    buildsystem: meson
+    builddir: true
+    config-opts:
+      - "-Dexamples=false"
+      - "-Dtests=false"
+    sources:
+      - type: git
+        url: https://gitlab.gnome.org/GNOME/libhandy.git
```
Comment (barthalion) [inline com.arteeh.Flasher.yml:42]: You should specify a commit or tag here.

File: com.arteeh.Flasher.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    sources:
+      - type: git
+        url: https://github.com/libusb/libusb.git
+  # Flashing software this app uses
+  - name: openocd
+    config-opts:
+      - --enable-stlink
+    sources:
+      - type: git
+        url: https://git.code.sf.net/p/openocd/code
```
Comment (barthalion) [inline com.arteeh.Flasher.yml:32]: You should specify a commit or tag here.

File: com.arteeh.Flasher.yml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  # Access to ST-Link usb device
+  - --device=all
+
+modules:
+  # OpenOCD with ST-Link requires libusb to be installed
+  - name: libusb
+    buildsystem: autotools
+    sources:
+      - type: git
+        url: https://github.com/libusb/libusb.git
```
Comment (barthalion) [inline com.arteeh.Flasher.yml:25]: You should specify a commit or tag here.

File: com.arteeh.Flasher.yml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  - --socket=wayland
+  # Download binaries from the internet
+  - --share=network
+  # Access to ST-Link usb device
+  - --device=all
+
+modules:
+  # OpenOCD with ST-Link requires libusb to be installed
+  - name: libusb
+    buildsystem: autotools
```
Comment (barthalion) [inline com.arteeh.Flasher.yml:22]:
```suggestion
```
It's the default buildsystem.

**Comment by barthalion** [comment]
> PNG is recommended, and the link from my metainfo does work in the browser. Am I supposed to host it somewhere without HTTPS?

I think it's going to work fine when the app is published to Flathub, just a bug somewhere in the test repo.

**Comment by arteeh** [comment]
bot, build com.arteeh.Flasher

**Comment by arteeh** [comment]
Hold on, the icon is broken.

**Comment by arteeh** [comment]
bot, build com.arteeh.Flasher

**Pull request closed without merging**